src/components/pages/MainQuizPage/EndQuizButton/EndQuizButton.tsx

```diff
@@ -1,8 +1,29 @@
+import { useState } from "react";
 import { StyledButton } from "../../../spreadedComponents/StyledButton/StyledButton";
+import { createPortal } from "react-dom";
+import { ModalPage } from "../../ModalPage/ModalPage";
+import { START } from "../../../Router/routes";
+import { useNavigate } from "react-router-dom";
 
 const EndQuizButton = () => {
+
+    const [showModal, setShowModal] = useState(false);
+    const navigate = useNavigate();
+
     return (
-        <StyledButton onClickHandler={() => { }} text="Finish test" />
+        <>
+            <StyledButton onClickHandler={() => setShowModal(true)} text="Finish test" />
+            {
+                showModal && createPortal(
```

Comment:
[recommendation]
if you're going to add animation for the modal appearing/disappearing I recommend you to remove `showModal` from this place and handle it inside the ModalPage component
otherwise you'll have to add timers and other things

Reply:
I can't remove showModal state into ModalPage and handle it there because this state is common for EndQuizButton component and ModalPage. Depends on value of this state I have different varieties of onClickHandler for "Finish button", because inside of this handler I use this state.
In my realisation when I push "Finish quiz" button at main page my application shows me modal window. But all functionality in main page continues, for example timer is going on or current active question is saved. That's why I don't fully understand why I should remove this state to ModalPage and how in this way I can show this modal window and remove back to main page with saved timer and active question.

Reply:
```tsx
function OtherMainComponent() {
   const [isShowModal, setIsShowModal] = useState(false)

  useEffect(() => {
    // check if time is over
    setIsShowModal(true)
  }, [])

  return (
    <div>
      some content here
      <ModalPage isOpen={isShowModal} onClose={setIsShowModal}>modal content</ModalPage>
    </div>
  )
}

function ModalPage({ isOpen, onClose, children }) {
  if (isOpen) return null
  
  return createPortal(
    <div>
      {children}
      <button onClick={onClose}>close</button>
    </div>,
    document.body
  )
}
```
with this approach you may reuse ModalPage + work with animation

Reply:
I see, and fixed this. But I didn't create useEffect as you coded above, because in our homework we should get to Modal Page only after pushing button. When timer will end at the main page we will get to ResultPage and this feature I realised in custom hook useInitialiseTimer and it works in a correct way